src/stdlib_io_disp.fypp

```diff
@@ -0,0 +1,125 @@
+#:include "common.fypp"
+#:set RIL_KINDS_TYPES = REAL_KINDS_TYPES + INT_KINDS_TYPES + LOG_KINDS_TYPES
+
+submodule (stdlib_io) stdlib_io_disp
+
+    use, non_intrinsic :: stdlib_strings, only: format_string
+    implicit none
+    character(len=*), parameter :: fmt_r = '(*(g11.4, 1x))'
+    character(len=*), parameter :: fmt_c = '(*(g23.4, 1x))'
+    
+contains
+    
+    #:for kind, type in RIL_KINDS_TYPES
+    module procedure disp_0_${type[0]}$${kind}$
+        if(present(string)) print *, trim(string)
+        print fmt_r, val
+    end procedure disp_0_${type[0]}$${kind}$
+
+    module procedure disp_1_${type[0]}$${kind}$
+        integer :: i, m
+        m = size(val)
+        if(present(string)) print *, trim(string)
+        print fmt_r, (val(i), i=1,m)
+    end procedure disp_1_${type[0]}$${kind}$
+
+    module procedure disp_2_${type[0]}$${kind}$
+        integer :: i, j, m, n
+        m = size(val, 1)
+        n = size(val, 2)
+        if(present(string)) print *, trim(string)
+        do i = 1, m
+            print fmt_r, (val(i,j), j=1, n)
```

Comment:
I think `print fmt_r, val(i,:)` just works, since an asterisk is already in `fmt_r = '(*(g11.4, 1x))'`.